File: leetcode_75/Prefix_Sum/python/724.py
from typing import List


class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        s = sum(nums)
        ps = 0
        for i in range(len(nums)):
            if ps == s - ps - nums[i]:
                return i
            ps += nums[i]

        return -1

File: leetcode_75/Prefix_Sum/cpp/724.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution
{
public:
    int pivotIndex(vector<int> &nums)
    {
        int s = 0;
        for (auto num : nums)
        {
            s += num;
        }

        int ps = 0;
        for (int i = 0; i < nums.size(); ++i)
        {
            if (ps == s - ps - nums[i])
            {
                return i;
            }
            ps += nums[i];
        }

        return -1;
    }
};